src/plugins/intel_cpu/src/nodes/kernels/x64/random_uniform.cpp

```diff
@@ -626,10 +640,675 @@ void RandomUniform<isa>::tail(const std::vector<Vmm>& vmm_dst) {
+    vpxord(v_result, v_result, v_aux);   // x ^= tmp
+}
+
+template <x64::cpu_isa_t isa> // Works for SSE41, AVX2
+void MersenneTwisterGenerator<isa>::generateRandomNumbers() {
+    const auto v_aux = getVmm();
+
+    // Load values from memory, copy
+    uni_vmovdqu(v_result, v_state);        // x = state
+
+    // Apply Mersenne Twister transformations
+    uni_vmovdqu(v_aux, v_result);           // tmp = x
+
+    // x ^= (x >> 11);
+    uni_vpsrld(v_aux, v_aux, 11);           // tmp >>= 11
+    uni_vpxor(v_result, v_result, v_aux);   // x ^= tmp
+
+    // x ^= (x << 7) & const_1;
+    uni_vmovdqu(v_aux, v_result);           // tmp = x
+    uni_vpslld(v_aux, v_aux, 7);            // tmp <<= 7
+    uni_vpand(v_aux, v_aux, v_const_1);     // tmp &= const_1
+    uni_vpxor(v_result, v_result, v_aux);   // x ^= tmp
+
+    // x ^= (x << 15) & const_2;
+    uni_vmovdqu(v_aux, v_result);           // tmp = x
+    uni_vpslld(v_aux, v_aux, 15);           // tmp <<= 15
+    uni_vpand(v_aux, v_aux, v_const_2);     // tmp &= const_2
+    uni_vpxor(v_result, v_result, v_aux);   // x ^= tmp
+
+    // x ^= (x >> 18);
+    uni_vmovdqu(v_aux, v_result);           // tmp = x
+    uni_vpsrld(v_aux, v_aux, 18);           // tmp >>= 18
+    uni_vpxor(v_result, v_result, v_aux);   // x ^= tmp
+}
+
+template <>
+void MersenneTwisterGenerator<x64::avx512_core>::convertToOutputTypeMersenne() {
+    if (m_jcp.out_data_type == element::f32) {
+        // Apply mask and divisor
+        // No need to do int32's voodoo with double since mask ensures
+        // that most significant bit is 0
+        vpandd(v_result, v_result, v_mask);
+        vcvtdq2ps(v_result, v_result);
+        vmulps(v_result, v_result, v_divisor);
+
+        // Scale and shift
+        vmulps(v_result, v_result, v_range);
+        vaddps(v_result, v_result, v_min);
+    } else if (m_jcp.out_data_type == element::f16) {
+        // Apply mask and divisor
+        vpandd(v_result, v_result, v_mask);
+        vcvtdq2ps(v_result, v_result);
+        vmulps(v_result, v_result, v_divisor);
+
+        vcvtps2ph(v_result, v_result, _MM_FROUND_TO_NEAREST_INT | _MM_FROUND_NO_EXC);
+
+        // View only half as v_result shrunk from 32->16 conversion
+        auto ymm_result = Xbyak::Ymm(v_result);
+        auto ymm_range = Xbyak::Ymm(v_range);
+        auto ymm_min = Xbyak::Ymm(v_min);
+
+        // Scale and shift
+        vmulph(ymm_result, ymm_result, ymm_range);
+        vaddph(v_result, ymm_result, ymm_min);
+    } else if (m_jcp.out_data_type == element::bf16) {
+        // Apply mask and divisor
+        vpandd(v_result, v_result, v_mask);
+        vcvtdq2ps(v_result, v_result);
+        vmulps(v_result, v_result, v_divisor);
+
+        // Pseudoconvert to f32 by bitshift
+        vpslld(v_range, v_range, 16);
+        vpslld(v_min, v_min, 16);
+
+        // Scale and shift
+        vmulps(v_result, v_range);
+        vaddps(v_result, v_min);
+
+        vcvtneps2bf16(v_result, v_result);
```

Comment:
How does it work on Cascade Lake for example? Need to check x64::mayiuse(x64::avx512_core_bf16) for bf16 and x64::mayiuse(x64::avx512_core_fp16) for fp16.